app/models/piece.rb

```diff
@@ -11,6 +11,29 @@ class Piece < ActiveRecord::Base
 
   # Instance Methods
 
+  def move_to(end_coord)
+
+    #Check if move is vaild for Piece
+    return false unless self.valid_move?(end_coord)
+
+    #Capture logic
+    game.uncaptured_pieces.each do |piece|
+      if end_coord == piece.coordinate
+        if self.color != piece.color
+          piece.destroy
```

Comment:
Do we want to destroy it or just mark it captured? Because if we destroy it, it will be harder to figure out what pieces are available if a pawn reaches the end of the board.